robonomics/io/src/source/virt.rs

```diff
@@ -59,6 +61,22 @@ pub fn pubsub(
     Ok(pubsub.subscribe(&topic_name).map(|v| Ok(v)))
 }
 
+/// Read data records from blockchain.
+///
+/// Returns datalog data objects.
+pub fn datalog(
+    remote: String,
+) -> Result<(
+    impl Sink<AccountId, Error = Error>,
```

Comment:
This one is a bit complex, let’s move address to function argument, it will make API much more simple